app/controllers/api/widgets_controller.rb

```diff
@@ -12,8 +12,7 @@ def generate_content_resource(type, id, _data = nil)
 
     def generate_widget_content(widget)
       desc = "#{widget_ident(widget)} content generation"
-      task_id = queue_object_action(widget, desc, :method_name => "generate_content", :args => ["User", User.current_user.current_group.description, [User.current_user.userid]])
-      action_result(true, desc, :task_id => task_id)
+      widget.queue_generate_content
```

Comment:
success action_result signature missing. should complement the failure, even without a task_id, i.e. action_result(true, desc)